ipatests/test_integration/test_vault.py

```diff
@@ -134,9 +140,36 @@ def test_create_and_retrieve_vault_replica_with_kra(self):
 
         self._retrieve_secret([self.vault_name_master2])
 
-        ################ old vaults ###############
+        '''################ old vaults ###############'''
         # test if old vaults are still accessible
         self._retrieve_secret([
             self.vault_name_master,
             self.vault_name_replica_without_KRA,
         ])
+
+    def test_vault_archive_overwrites_existing_value_with_warning(self):
+        self.master.run_command([
+            "ipa", "vault-add",
+            self.vault_name_master4,
+            "--password", self.vault_password,
+        ])
+
+        # archive secret
+        self.master.run_command([
+            "ipa", "vault-archive",
+            self.vault_name_master4,
+            "--password", self.vault_password,
+            "--data", self.vault_data,
+        ])
+        time.sleep(WAIT_AFTER_ARCHIVE)
+
+        self._retrieve_secret([self.vault_name_master4])
+
+        result = self.master.run_command([
+            "ipa", "vault-archive",
+            self.vault_name_master4,
+            "--password", self.vault_password,
+            "--data", self.vault_data2,
+        ])
+        time.sleep(WAIT_AFTER_ARCHIVE)
+        assert 'OVERWRITE_WARN_MESG' in result.stdout_text
```

Comment:
you are comparing string 'OVERWRITE_WARN_MESG' with actual output instead of checking that a content of a variable `OVERWRITE_WARN_MESG` is there. Please change this to

```
  assert OVERWRITE_WARN_MESG in result.stdout_text
```